### DISCUSSION TITLE
Footnotes created in a loop - article contributors/corresponding author

### DISCUSSION THREAD

**Discussion by arnold-c** [typst/typst#4611] [Q&A] [answered]
I'm very new to Typst and enjoying the relative ease as creating and editing templates, but I'm stuck trying to get footnotes to work correctly for an article template I'm creating. I've been modifying [this template](https://github.com/typst/typst/discussions/1504) to add things like equal contribution footnotes and noticed that the numbering in the footnotes don't match numbering in the text. I suspect it's a scoping issue related to the for loop that cycles through all authors , but I'm not sure how to fix that issue. This issue persists in the rest of the document as well (see page 2)

One thought I had was to try and create a footnote numbering system that only provides one option, but while I could get that to work outside of a footnote call, it seems like the `footnote` function has some restrictions on the `numbering` system that can be implemented (see the end of page 2)

The document can be [found here](https://typst.app/project/rbtMsyRZKgn5LrXVWzMcCq)

Thanks!

**Comment by EpicEricEE** [comment]
Late reply, but you seem to be confusing the footnote _number_ with the footnote _numbering_. The numbering only defines how a number should be displayed and does not contain any information about the number itself. For example, the numbering `"a"` defines that the number `3` should be displayed as `"c"`.

You have this at the top of your template.
```typ
#show footnote.entry: it => [
  #set par(hanging-indent: 0.7em)
  #set align(left)
  #it.note.numbering #it.note.body
]
```
By using only `it.note.numbering`, you don't display the actual number of the footnote, but only its numbering. What you instead want to use, is `counter(footnote).display(it.note.numbering)`, which means that the current footnote number should be displayed with the footnote's numbering.

The problem you have at the end of the document with the "fifth footnote" is that `sym.dagger` is not a valid numbering pattern and thus cannot be used as a value for the `numbering` parameter. It seemingly works outside the footnote call, because when you call the `numbering` function with `n => values.at(0)`, you say that every number should get converted to `sym.dagger`. As the result of this function is always `sym.dagger` (still not a valid numbering pattern), you cannot use this function call as a parameter. You can find all valid numbering patterns [in the docs](https://typst.app/docs/reference/model/numbering/#parameters-numbering) under "Counting symbols".

**Reply by arnold-c** [reply, marked as answer]
Thank you very much for the explanation, and not a problem about the late reply - I've been away for the last month, so it wasn't a delay from my end!

I updated it to use the below, but ran into a different issue that all the numbers of the footnotes were the same (though they showed the correct bodies) i.e.,

```typ
#show footnote.entry: it => [
  #set par(hanging-indent: 0.7em)
  #set align(left)
  #counter(footnote).display(it.note.numbering) #it.note.body
]
```

produced

![image](https://github.com/user-attachments/assets/1430d6ed-4374-4362-ab4c-a375f37c2972)

> Note: I have replaced the numbering with "a" as it's a little easier to track what's going on

Using `.get()` shows that for some reason, the same counter number is being reused for all the footnotes i.e.,

```typ
#show footnote.entry: it => [
  #set par(hanging-indent: 0.7em)
  #set align(left)
  #context counter(footnote).get() counter(footnote).display(it.note.numbering) #it.note.body
]
```

produced

![image](https://github.com/user-attachments/assets/cbbd0f42-78f5-4d33-95ff-971c4e9c1da1)


The solution (I believe) is to simply use `#it` to return the correct footnote and body i.e.,

```typ
#show footnote.entry: it => [
  #set par(hanging-indent: 0.7em)
  #set align(left)
  #it
]
```

![image](https://github.com/user-attachments/assets/04d527a6-6626-4b8f-9a0a-a6cfb0d6a16f)

Though, if this is incorrect and will cause an issue I haven't come across yet, please do point it out!

Thanks for the help.

**Reply by EpicEricEE** [reply]
> the same counter number is being reused for all the footnotes

Yes, you're right, that was a mistake on my side. The counter is stepped when the footnote is placed in the text and the entries are placed afterwards. When the entries are placed, the counter has already reached the final number. It should have been
```typ
#numbering(it.note.numbering, ..counter(footnote).at(it.note.location()))
```
to get the correct number. What you have now done is also correct (with the difference that the other solution would not have used superscripts for the footnote number), though it would be more idiomatic to use [show-set rules](https://typst.app/docs/reference/styling/#show-rules)
```typ
#show footnote.entry: set par(hanging-indent: 0.7em)
#show footnote.entry: set align(left)
```

**Reply by arnold-c** [reply]
Thanks for the update. That's exactly what I was hoping to do.
